src/core/componentSystems/BoundaryVisibilitySystem.js

```diff
@@ -0,0 +1,81 @@
+import { MRSystem } from 'mrjs/core/MRSystem';
+import { MRDivEntity } from 'mrjs/core/MRDivEntity';
+import { MRPanel } from 'mrjs/core/entities/MRPanel';
+
+/**
+ * @function
+ * @description Observe a target MRDivEntity and make the associated object visible only if it is in visible position in a root MRDivEntity
+ * @param {MRDivEntity} root
+ * @param {MRDivEntity} target
+ */
+const observe = (root, target) => {
+    // TODO: Callback is fired asynchronously so no guaranteed to be called immediately when the
+    //       visibility from the layout position changes. Therefore, the visibility of the associated
+    //       Object3D's might be updated a few frames later after when it really need to be. It might
+    //       affect the user experience. Fix it if possible.
```

Comment:
I'm happy if you folks test on headset and share thoughts @michaelthatsit @hanbollar

Reply:
I think that delay is fine for now - as long as fixing the order per [link](https://github.com/Volumetrics-io/mrjs/pull/516/files#r1536268465) lets things run smoothly (ie the other systems can override / run what they need to) then it should be okay